src/Compilers/CSharp/Portable/Lowering/LambdaRewriter/SynthesizedClosureMethod.cs

```diff
@@ -16,6 +16,7 @@ internal sealed class SynthesizedClosureMethod : SynthesizedMethodBaseSymbol, IS
     {
         private readonly MethodSymbol _topLevelMethod;
         private readonly ImmutableArray<NamedTypeSymbol> _structEnvironments;
+        private readonly bool _areLocalsZeroed;
```

Comment:
Eventually we want the bit propagate form any enclosing context, not just method. 
That would likely move you towards having this bit as one of `BinderFlags`. Then that solution will be more general and will handle this case as well. 

I wonder if we should just add AreLocalsZeroed on the binder and propagate via the `BinderFlags` ?

Reply:
I thought about this some more.

I see two possible approaches: store on the symbol and ask ContainingSymbol, or store in the binder. I see these approaches as very similar and not needed to progress on the PR.

My idea is this: what we really want to do is have the SynthesizedClosureMethod ask its BaseMethod for AreLocalsZeroed. This means that the LocalFunction or Lambda would return true or false depending on their implementation.

The LocalFunction and Lambda implementations would, in my mind, ask their containing symbol or return an answer, if they have one. This can be extended to any symbol, which can always ask its containing symbol.

This could be stored on the binder instead, but I think deciding whether to ask the containing symbol or ask the binder is an orthogonal change, which we can switch however we like.